### PR TITLE
AArch64: Implement ArrayStoreCHKEvaluator()

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements ArrayStoreCHKEvaluator() for AArch64.

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
What does the `generateMemSrc1Instruction()` in this evaluator store, and what is its destination?
I brought the code from the arm32 and p versions and modified it for AArch64, but I haven't understood it enough yet.

**Comment by knn-k** [comment]
Resolved the merge conflict.

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -356,4 +358,76 @@ J9::ARM64::TreeEvaluator::BNDCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
...
+   cg->machine()->setLinkRegisterKilled(true);
+   }
+
+static void VMoutlinedHelperWrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register *dstReg, TR::CodeGenerator *cg)
+   {
+   auto gcMode = TR::Compiler->om.writeBarrierType();
+   if (gcMode == gc_modron_wrtbar_none)
+      return;
+
+   TR::Compilation *comp = TR::comp();
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp]: Get the Compilation object from `cg`:   `cg->comp()`

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 2]
```diff
@@ -356,4 +358,76 @@ J9::ARM64::TreeEvaluator::BNDCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
...
+   gcPoint->ARM64NeedsGCMap(cg, 0xFFFFFFFF);
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, doneLabel, deps);
+
+   cg->machine()->setLinkRegisterKilled(true);
+   }
+
+TR::Register *
+J9::ARM64::TreeEvaluator::ArrayStoreCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp]: To make this easier to read, can you store `node->getFirstChild()` into a `wrtBarNode` and use that instead?

**Comment by 0xdaryl** [comment]
> What does the generateMemSrc1Instruction() in this evaluator store, and what is its destination?
I brought the code from the arm32 and p versions and modified it for AArch64, but I haven't understood it enough yet.

That instruction actually does the store of the object into the array.  Its destination is the element address obtained from the first child of the write barrier node (i.e, the ASC's first child).  This store is only permitted if the type checks succeed.  It must also be followed with a write barrier.

**Comment by knn-k** [comment]
Reflected review comments, and resolved merge conflicts.

**Comment by 0xdaryl** [comment]
AArch64-specific changes.  No Jenkins required.

**Pull request merged**